### PR TITLE
Update core to PHP 7.4 standard

### PR DESCRIPTION
**Pull request by CarlSchwan**
- Typed properties
- Port to LoggerInterface

Signed-off-by: Carl Schwan <carl@carlschwan.eu>

### CODE REVIEW

File: core/Command/App/Update.php [hunk 1]
```diff
@@ -30,26 +30,19 @@
 use OC\Installer;
 use OCP\App\IAppManager;
 use OCP\ILogger;
```
Comment (artonge) [inline core/Command/App/Update.php:32]: Remove?

File: core/Command/Encryption/ChangeKeyStorageRoot.php [hunk 2]
```diff
@@ -29,6 +29,7 @@
 use OC\Encryption\Util;
 use OC\Files\Filesystem;
 use OC\Files\View;
+use OC\URLGenerator;
```
Comment (artonge) [inline core/Command/Encryption/ChangeKeyStorageRoot.php:32]: Not sure why it's added

File: core/Controller/AvatarController.php [hunk 3]
```diff
@@ -41,54 +41,36 @@
...
 								IRequest $request,
 								IAvatarManager $avatarManager,
 								ICache $cache,
 								IL10N $l10n,
 								IUserManager $userManager,
 								IRootFolder $rootFolder,
-								ILogger $logger,
-								$userId,
+								LoggerInterface $logger,
+								?string $userId,
```
Comment (artonge) [inline core/Controller/AvatarController.php:73]: Why optional?

File: core/Controller/ClientFlowLoginController.php [hunk 4]
```diff
@@ -122,30 +96,20 @@ public function __construct($appName,
...
 		$userAgent = $this->request->getHeader('USER_AGENT');
 		return $userAgent !== '' ? $userAgent : 'unknown';
 	}
 
-	/**
-	 * @param string $stateToken
-	 * @return bool
-	 */
-	private function isValidToken($stateToken) {
+	private function isValidToken($stateToken): bool {
```
Comment (artonge) [inline core/Controller/ClientFlowLoginController.php:104]:
```suggestion
	private function isValidToken(string $stateToken): bool {
```

File: core/Controller/SearchController.php [hunk 5]
```diff
@@ -32,19 +32,17 @@
 use OCP\IRequest;
 use OCP\ISearch;
 use OCP\Search\Result;
+use Psr\Log\LoggerInterface;
```
Comment (artonge) [inline core/Controller/SearchController.php:35]: `use OCP\ILogger;` was not removed

File: core/Controller/TwoFactorChallengeController.php [hunk 6]
```diff
@@ -39,35 +39,17 @@
 use OCP\ISession;
 use OCP\IURLGenerator;
 use OCP\IUserSession;
+use Psr\Log\LoggerInterface;
```
Comment (artonge) [inline core/Controller/TwoFactorChallengeController.php:42]: `use OCP\ILogger;` was not removed

File: core/Controller/AvatarController.php (hunk 3 above)
Comment (artonge) [reply core/Controller/AvatarController.php:73]: Should this be reflected on line 62?

**Comment by CarlSchwan** [comment]
found some issues, don't merge

**Review by blizzz — CHANGES_REQUESTED** [review]
cd. https://github.com/nextcloud/server/pull/31948#issuecomment-1097930067

**Review by come-nc — COMMENTED** [review]
Please rebase and fix or explain the issues

**Comment by CarlSchwan** [comment]
> found some issues, don't merge

Fixed the issues found last time and all the unit tests now pass (needed to change from ILogger to LoggerInterface there too)

File: core/Command/App/Update.php [hunk 7]
```diff
@@ -114,7 +106,11 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 					try {
 						$result = $this->installer->updateAppstoreApp($appId, $input->getOption('allow-unstable'));
 					} catch (\Exception $e) {
-						$this->logger->logException($e, ['message' => 'Failure during update of app "' . $appId . '"','app' => 'app:update']);
+						$this->logger->error($e->getMessage(), [
+							'message' => 'Failure during update of app "' . $appId . '"',
+							'app' => 'app:update',
+							'exception' => $e,
+						]);
```
Comment (come-nc) [inline core/Command/App/Update.php:113]:
```suggestion
						$this->logger->error('Failure during update of app "' . $appId . '"', [
							'app' => 'app:update',
							'exception' => $e,
						]);
```

File: lib/private/Repair/NC22/LookupServerSendCheck.php [hunk 8]
```diff
@@ -33,12 +33,8 @@
...
 
 class LookupServerSendCheck implements IRepairStep {
-
-	/** @var IJobList */
-	private $jobList;
-
-	/** @var IConfig */
-	private $config;
+	private IJobList$jobList;
+	private IConfig$config;
```
[the remark this answers is not in the document]
Comment (come-nc) [reply lib/private/Repair/NC22/LookupServerSendCheck.php:37]: Maybe cs:fix would fix those?

**Review by come-nc — CHANGES_REQUESTED** [review]

File: core/Command/Encryption/DecryptAll.php [hunk 9]
```diff
@@ -181,13 +160,12 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
-				$this->config->setAppValue('core', 'encryption_enabled', 'yes');
-				$output->writeln('done.');
-				$output->writeln('aborted');
-				return 1;
 			}
+			$output->write('Enable server side encryption... ');
+			$this->config->setAppValue('core', 'encryption_enabled', 'yes');
+			$output->writeln('done.');
+			$output->writeln('aborted');
+			return 1;
```
[the remark this answers is not in the document]
Comment (CarlSchwan) [reply core/Command/Encryption/DecryptAll.php:168]: To me this is more readable. There is no need for an else block if we already return in the if bloc. This makes the code less nested and more linear.

**Review by come-nc — APPROVED** [review]

**Comment by come-nc** [comment]
@artonge @blizzz Should be good for merging now?

**Review by artonge — APPROVED** [review]

**Pull request merged**